### DISCUSSION TITLE
Building a menubar

### DISCUSSION THREAD

**Discussion by clarknova** [floating-ui/floating-ui#2517] [Q&A] [answered]
I want to building a menubar like this: https://www.radix-ui.com/primitives/docs/components/menubar
I am having a hard time. I tried to adapt this example https://codesandbox.io/s/admiring-lamport-5wt3yg?file=/src/DropdownMenu.tsx The Problem is that the main menubar is not a floated element  but i still want all the elements in there to participated in the keyboard navigation etc.

**Comment by atomiks** [comment]
You can use the list navigation hook on regular non-floating elements by setting `open: true` permanently in `useFloating`.  Everything will then work as normal even though the menu bar is not actually a floating element.

But yes, the list navigation hook needs to be split out into better primitives for this to work better. 

Issue: https://github.com/floating-ui/floating-ui/issues/2497

**Comment by atomiks** [comment, marked as answer]
The new `Composite` components in the latest version allow list navigation to create a menubar: https://floating-ui.com/docs/Composite
